### PR TITLE
feat: Replace the current TOC implementation with the element from the payload

### LINKED ISSUE #108: Problem: table of content is redundant in some articles
Problem: This article is way too small and TOC is redundant https://holdex.io/c/companies/clearpool-finance
Solution: make a table of content a component of the thread body (like chart, image, etc.), so we can decide if we want to have it and where we want to have it.

Requested solution
1. Update `api/google-conversion` file to parse Google Document JSON (e.g. https://api.npoint.io/63d3b4aaa5cd23361e65), and return custom element `type: toc`
2. replace the current TOC implementation with the element from the payload

### PR DESCRIPTION
**Pull request by accompany1205**
resolves #108 
ETA: 4 hrs
Time taken so far: 5 hrs

### CODE REVIEW

File: src/lib/components/BodyParser/index.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ class Parser {
       isGoogleDoc,
       blocks,
       parsedBody,
-      tocs,
+      tocs: tocs.data,
```
Comment (georgeciubotaru) [inline src/lib/components/BodyParser/index.ts:54]: what if tocs are null?

File: src/lib/components/BodyParser/index.ts [hunk 2]
```diff
@@ -41,8 +41,8 @@ class Parser {
       parsedBody?.blocks
     );
     const blocks = Parser.parseBlocks(parsedBlocks);
-    const tocs = Parser.parseTocs(blocks);
     const cover = Parser.parseThreadCover(blocks);
+    const tocs = blocks.find((block) => block.type === 'toc');
```
Comment (georgeciubotaru) [inline src/lib/components/BodyParser/index.ts:45]: remove the old function is not used

File: src/lib/components/BodyParser/index.ts (hunk 1 above)
Comment (accompany1205) [reply src/lib/components/BodyParser/index.ts:54]: If tocs is null, it doesn't show
image

**Comment by georgeciubotaru** [comment]
@accompany1205 URL preview doesn't work for me

**Comment by accompany1205** [comment]
@georgeciubotaru Do you mean Vercel url?
Could you redeploy again?

**Comment by georgeciubotaru** [comment]
@accompany1205 I mean the preview URL of this PR.

I don't need to redeploy as you can push changes to Vercel without consent

**Comment by accompany1205** [comment]
@georgeciubotaru What should I do?

**Comment by georgeciubotaru** [comment]
@accompany1205 Please test locally, not only the thread from screenshot, try to access other pages and check if all good.

**Comment by accompany1205** [comment]
@georgeciubotaru I tested other pages locally, it works.
And Visit Preview is working now

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**